include/crow/middlewares/cors.h

```diff
@@ -0,0 +1,159 @@
+#pragma once
+#include "crow/http_request.h"
+#include "crow/http_response.h"
+
+namespace crow
+{
+    struct CORSHandler;
+
+    // CORSRules is used for tuning cors policies
+    struct CORSRules
+    {
+        friend struct crow::CORSHandler;
+
+        // Set Access-Control-Allow-Origin. Default is "*"
+        CORSRules& origin(const std::string& origin)
+        {
+            origin_ = origin;
```

Comment:
We could elide copies here by making the code more complex. Given that this is called once on startup, I don't think its worth it